modules/tinymce/src/plugins/imagetools/main/ts/core/Actions.ts

```diff
@@ -20,43 +21,52 @@ import * as ImageSize from './ImageSize';
 
 let count = 0;
 
-const getFigureImg = (elem) => SelectorFind.child(SugarElement.fromDom(elem), 'img');
+const enum FileExtractType {
+  Name = 2,
+  NameExt = 1
+}
 
-const isFigure = (editor: Editor, elem) => editor.dom.is(elem, 'figure');
+const getFigureImg = (elem: HTMLElement) => SelectorFind.child<HTMLImageElement>(SugarElement.fromDom(elem), 'img');
 
-const getEditableImage = (editor: Editor, elem) => {
-  const isImage = (imgNode) => editor.dom.is(imgNode, 'img:not([data-mce-object],[data-mce-placeholder])');
-  const isEditable = (imgNode) => isImage(imgNode) && (isLocalImage(editor, imgNode) || isCorsImage(editor, imgNode) || Settings.getProxyUrl(editor));
+const isFigure = (editor: Editor, elem: Node): elem is HTMLElement => editor.dom.is(elem, 'figure');
 
-  if (isFigure(editor, elem)) {
-    const imgOpt = getFigureImg(elem);
-    return imgOpt.map((img) => isEditable(img.dom) ? Optional.some(img.dom) : Optional.none());
```

Comment:
This was a minor bug discovered by adding the types, as the return type needed to be `Optional<HTMLImageElement>` but this logic returned `Optional<Optional<HTMLImageElement>>`.